### PR TITLE
[master][IMP] stock: add hook to control lot creation on incoming shipments

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
Nothing changes functionally with this PR, it is just a hook to improve extensibility of the code and control the lot creation from stock move lines.

@ForgeFlow

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by LoisRForgeFlow** [comment]
@amoyaux Could you have a look? Thanks!

File: addons/stock/models/stock_move_line.py [hunk 1]
```diff
@@ -594,6 +594,14 @@ def _get_similar_move_lines(self):
             lines |= picking_id.move_line_ids.filtered(lambda ml: ml.product_id == self.product_id and (ml.lot_id or ml.lot_name))
         return lines
 
+    def _prepare_new_lot_vals(self):
+        self.ensure_one()
+        return {
+            'name': self.lot_name,
+            'product_id': self.product_id.id,
+            'company_id': self.move_id.company_id.id,
```
Comment (amoyaux) [inline addons/stock/models/stock_move_line.py:602]: no need to check on the move anymore

**Comment by amoyaux** [comment]
Ok otherwise

File: addons/stock/models/stock_move_line.py (hunk 1 above)
Comment (LoisRForgeFlow) [reply addons/stock/models/stock_move_line.py]: Fixed.

**Comment by LoisRForgeFlow** [comment]
@amoyaux I just attended your comment. Could you have a look again?

**Comment by amoyaux** [comment]
@LoisRForgeFlow that's good for me. Thanks for the contribution

**Comment by amoyaux** [comment]
robodoo r+

**Pull request closed without merging**